test/uk/gov/hmrc/alcoholdutyreturns/controllers/CacheControllerSpec.scala

```diff
@@ -101,22 +119,37 @@ class CacheControllerSpec extends SpecBase {
   }
 
   "add" should {
-    "return 200 OK with the user answers that was inserted and the account service return a UserAnswers" in {
-      when(mockCacheRepository.add(any())).thenReturn(Future.successful(true))
-      when(mockAccountService.createUserAnswers(any())(any(), any())).thenReturn(EitherT.rightT(emptyUserAnswers))
+    "return 200 OK with the user answers that was inserted" when {
+      "the account service returns a valid UserAnswers" in {
+        when(mockCacheRepository.add(any())).thenReturn(Future.successful(userAnswers))
+        when(mockAccountService.createUserAnswers(any())(any(), any())).thenReturn(EitherT.rightT(userAnswers))
 
-      val result: Future[Result] =
-        controller.add()(
-          fakeRequestWithJsonBody(Json.toJson(emptyUserAnswers))
-        )
+        val result: Future[Result] =
+          controller.add()(
+            fakeRequestWithJsonBody(Json.toJson(emptyUserAnswers))
+          )
 
-      status(result)        shouldBe OK
-      contentAsJson(result) shouldBe Json.toJson(emptyUserAnswers)
+        status(result)        shouldBe OK
+        contentAsJson(result) shouldBe Json.toJson(userAnswers)
+      }
+
+      "the account service returns a UserAnswers with bad data" in {
```

Comment:
A consequence of having coverage set to 100% - the warn being logged (can't get the data back we expect) can't be tested. At best this shows the service still works.